src/libraries/Microsoft.PowerFx.Core/Public/Types/TableType.cs

```diff
@@ -33,7 +34,22 @@ public override void Visit(ITypeVistor vistor)
 
         public TableType Add(NamedFormulaType field)
         {
+            var displayNameProvider = _type.DisplayNameProvider;
+            if (field.DisplayName != default)
+            {
+                if (displayNameProvider == null)
+                    displayNameProvider = new DisplayNameProvider();
+
+                displayNameProvider = displayNameProvider.AddField(field.Name, field.DisplayName);
+            }
+
             var newType = _type.Add(field._typedName);
+
+            if (displayNameProvider != null)
+            {
+                newType = DType.ReplaceDisplayNameProvider(newType, displayNameProvider);
+            }
```

Comment:
This looks the same as the new logic in RecordType.Add. 
Can this be shared and moved to AggregateType base class?